libr/anal/var.c

```diff
@@ -68,11 +68,9 @@ R_API bool r_anal_var_display(RAnal *anal, int delta, char kind, const char *typ
 	return true;
 }
 
-R_API bool r_anal_var_add(RAnal *a, ut64 addr, int scope, int delta, char kind, const char *type, int size,
-		bool isarg, const char *name) {
-	if (!a) {
-		return false;
-	}
+R_API bool r_anal_var_add(RAnal *a, ut64 addr, int scope, int delta, char kind, R_IFNULL("int") const char *type, int size, bool isarg, R_NONNULL const char *name) {
```

Comment:
shall we add `R_IFNULL(R_ANAL_VAR_KIND_BPV) char kind` too? maybe we need to split the signature in 2 lines since it's becoming long :/

Reply:
The function signature is too complex imho, we should rethink it maybe. But having it in one line is the only sane way to make it work with grep.

R_IFNULL(kind) makes no sense because char cant be null, unless we assume that a nul char is the same as null. but it looks inconsistent to me to use R_IFNULL with char.

Reply:
You are right.